sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -572,8 +572,9 @@ bool _pi_queue::isBatchingAllowed() {
 }
 
 pi_result _pi_queue::batchCommandList(ze_command_list_handle_t ZeCommandList,
-                                      ze_fence_handle_t ZeFence) {
-  if (this->isBatchingAllowed()) {
+                                      ze_fence_handle_t ZeFence,
+                                      bool OKToBatchKernel) {
+  if (OKToBatchKernel && this->isBatchingAllowed()) {
```

Comment:
Instead of adding the new `OKToBatchKernel` argument, could we move this 'if' out of this helper routine to its only caller and do the `executeCommandList` there.

Reply:
I thought about this, but then the code in piEnqueueKernelLaunch looks like:
if (OKToBatchKernel)
  res = batch
else
 res = execute

I much prefer hiding that inside the batch program, which is what is currently being done by the code.  This also makes it so that other places can use.  An alternative that I considered was moving the batching logic into executeCommandList, and just passing that boolean in there, and deleting batchCommandList entirely.  I would MUCH prefer to do that than to move the OKtoBatchKernel logic out into piEnqueueKernelLaunch.  I could also just change the name of this parameter to OKToBatchThisCommand, and again, either add that to batchCommandList or executeCommandList.

Reply:
I like what you show it would look like in piEnqueueKernelLaunch, this also makes the methods more focused on what they are aimed to do. 

> An alternative that I considered was moving the batching logic into executeCommandList, and just passing that boolean in there, and deleting batchCommandList entirely.

Sounds good too.

Reply:
Done